### DISCUSSION TITLE
Extending core entity of modules in medusa 2.0

### DISCUSSION THREAD

**Discussion by imash96** [medusajs/medusa#6807] [Q&A]
hello,

I'm thinking of moving to Medusa 2.0 and want to try out Medusa commerce modules like products, cart, pricing, etc.

There is a lot of customization in my existing project and I have created some new entity, as well as extending my core entitys like Product, Cart, LineItem, Order.

My question is how can I create and extended these entity, as well as extend custom fields and extend services, using the Medusa 2.0 Commerce modules.

Thank you

**Comment by alsherif-khalaf** [comment]
I think relying on metafields and widget UI is safer in the situation of upgrading

**Reply by imash96** [reply]
This is not a good solution

**Comment by imash96** [comment]
I think medusa 2.0 is more complicated and not well documented compare to medusa v1.x and other frameworks
